.pre-commit-config.yaml

```
@@ -0,0 +1,34 @@
+minimum_pre_commit_version: "2.9.0"
+repos:
+  - repo: https://github.com/Zac-HD/shed
+    rev: 0.9.5
+    hooks:
+      - id: shed
+        # TODO: bump to --py310-plus when Talon moves to Python 3.10.
+        args: [--refactor, --py39-plus]
+        types_or: [python, markdown, rst]
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+        # Trailing whitespace breaks yaml files if you use a multiline string
+        # with a line that has trailing white space. Many of our recorded
+        # tests use strings with trailing white space to represent the final
+        # document contents. For example
+        # src/test/suite/fixtures/recorded/languages/ruby/changeCondition.yml
+        exclude: ^src/test/suite/fixtures/recorded/.*/[^/]*\.yml$
+      - id: mixed-line-ending
+      - id: check-merge-conflict
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: "v8.13.0"
+    hooks:
+      - id: eslint
```

Might need to specify `--fix` to let eslint autofix if that's what you're after.
```suggestion
      - id: eslint
        args: ['--fix']
```

You might need to add `eslint-config-prettier` to `additional_dependencies` as well?

Looking at the [mirrors-eslint readme](https://github.com/pre-commit/mirrors-eslint#using-eslint-with-pre-commit) it looks like you might need to specify the file types for TypeScript?

---

Done thanks!